src/app/shared/classes/report-hierarchy-transformer.ts

```diff
@@ -62,20 +65,12 @@ export class ReportHierarchyTransformer {
     parentStartpoint.checkpoints.push(checkpoint);
   }
 
-  getImage(type: CheckpointType, encoding: string, level: number): string {
-    const even: boolean = this.determineEvenCheckpoint(level);
-    let img = `assets/tree-icons/${CHECKPOINT_TYPE_STRINGS[type]}`;
+  getImage(type: CheckpointType, encoding: string, level: number): IconData {
+    const iconData: IconData = { ...CHECKPOINT_TYPE_STRINGS[type] };
```

Comment:
`CHECKPOINT_TYPE_STRINGS` already returns an object of type IconData right?
```suggestion
    const iconData: IconData = CHECKPOINT_TYPE_STRINGS[type];
```

Reply:
The fields in the CHECKPOINT_TYPE_STRINGS will be in a mutable state if I do it without. I had a problem where it would keeping adding -odd and -even at the end, resulting in something like tree-checkpoint-odd-odd-odd-odd

Reply:
Ahh yeah I understand, it'se because you return the same object everytime, using the spread operator is actually fine if you need a shallow copy.

Reply:
I tried putting the const as a readonly type, but for some reason it would still be in a shared mutable state